### ISSUE TITLE
Tilemap bugs

### ISSUE THREAD

**Issue by loudoweb** [openfl/openfl#1192]
openfl 4.0.1, lime 3.0.0

**All targets except flash :** 

1. nothing is drawn when there is only one Tile in Tilemap
2. setting different tilesets in tiles is not working (it seems that only the bitmapData of the first Tileset is used)

flash target is working as expected in my current test.

**Comment by loudoweb** [comment]
if only one Tile, cacheBitmapData is null, so there is no bindTexture for this single Tile.

[Here ](https://github.com/openfl/openfl/blob/develop/openfl/_internal/renderer/opengl/GLTilemap.hx#L207-L219)we should add this line 208 : 
`if (i == count - 1)
					cacheBitmapData = tileset.bitmapData;`

It will allow to fix bug 1.

**Comment by jgranick** [comment]
Could you give this a try? https://github.com/openfl/openfl/commit/e8812c585eb0607084839c27475bc88ff64725b8

I think this only affected the OpenGL Tilemap implementation, there was a gap in the logic that I think this commit has fixed

**Comment by loudoweb** [comment]
It fixes 1. :+1: 

It's really better for 2. 
For example, if I use 2 atlas, it seems ok. But if I use 3 different atlas for 3 tiles, I can see a tiny image of the second texture onto the second tile. Let me few moment to make a screenshot.

**Comment by loudoweb** [comment]
Hello again!

Here is a screenshot of the issue : 

![Tilemap bug](http://img15.hostingpics.net/pics/822851tilemapbug.png)

Description of the problem : 

-there are 2 different atlases (one with brown wabbit (atlas A), other with green horizontal wabbit (atlas B))
-add a tile of atlas A, then a tile of B, then a tile of A again
-we see that on the third tile wich uses atlas A some artifacts of atlas B

**It means the bug occurs when we switch back to an already used atlas.**

I think we only see the bug if the current tile and the previous tile are not using the same rectangle, because in my first test where I used atlases with just different colors I didn't see the bug. But now my second atlas uses horizontal wabbit instead of vertical, the bug occurs.

Here are the atlases for your test.

[atlas.zip](https://github.com/openfl/openfl/files/375612/atlas.zip)

**Comment by jgranick** [comment]
Thanks! Can you please try the latest development version of Lime and OpenFL, and see how it works for you now? It works in this case, and I believe will work fine in larger tests, too, but it helps to have the confirmation :smile:

**Comment by loudoweb** [comment]
Thank you very much, it has been fixed! :+1: 

I've made some additional tests on tilemap. There are some bad rendering with rotation.

tiles rotation from left to right : 0, 45, 22, 10
![Tilemap bug](http://img15.hostingpics.net/pics/226319tilemapbug2.png)

(so the rendering is ok for a rotation of 0 or 45 degrees but not for 22 or 10 degrees)

Also when I made other tests with 2 atlases, an error occured on GLTilemap.hx line 78 (Error.OutsideBounds) but I don't know if I did something wrong or if something need to be fixed on GLTilemap.
